Wikipedia/Code/MWKDataStore.m

```diff
@@ -195,18 +197,25 @@ - (void)syncDataStoreToDatabase {
 }
 
 - (void)cleanup {
-    [self.writeConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *_Nonnull transaction) {
-        YapDatabaseViewTransaction *view = [transaction ext:WMFNotInHistorySavedOrBlackListSortedByURLUngroupedView];
-        if ([view numberOfItemsInAllGroups] == 0) {
-            return;
-        }
-        NSMutableArray *keysToRemove = [NSMutableArray array];
-        [view enumerateKeysInGroup:[[view allGroups] firstObject]
-                        usingBlock:^(NSString *_Nonnull collection, NSString *_Nonnull key, NSUInteger index, BOOL *_Nonnull stop) {
-                            [keysToRemove addObject:key];
-                        }];
-        [transaction removeObjectsForKeys:keysToRemove inCollection:[MWKHistoryEntry databaseCollectionName]];
-    }];
+    id previousCleanup = self.previousCleanup;
+    if (previousCleanup != nil) {
+        [NSObject bk_cancelBlock:previousCleanup];
```

Comment:
@coreyfloyd this is the intention. the idea is to debounce the cleanup so that it doesn't happen on every database change. this way, it should keep the cleanup from happening until the database is at rest for a second.